### ISSUE TITLE
Limit(x,x,a).free_symbols should return set([a])

### ISSUE THREAD

**Issue by mcho421** [sympy/sympy#9205]
```
>>> x,a = symbols('x a')
>>> Limit(x,x,a).free_symbols
set([+, a, x])
```

I think it should just return 'a', or at least not return the '+'.

**Comment by debugger22** [comment]
Since it is unevaluated, it returns both `x` and `a`. I think `+` is coming from `a+`, probably not expected.
```
In [14]: Limit(x,x,a).doit().free_symbols
Out[14]: set([a])
```

**Comment by mcho421** [comment]
I had the impression it would work like unevaluated integrals:

```
>>> Integral(x, (x, 1,2)).free_symbols
set([])
```

**Comment by debugger22** [comment]
Oh. Then it should be more consistent.

**Comment by asmeurer** [comment]
Yes, this should work. 

I also don't like how `Symbol('+')` is in the args. It would be cleaner to just have LeftLimit and RightLimit classes to distinguish.

**Comment by gxyd** [comment]
From Documentation
"""free_symbols
    Return from the atoms of self those which are free symbols."""

From this definition:
```
>>> p1 = Limit(x, x, a).atoms()
{+,a,x}              # checking the type reveals all are free_symbols in this set
                        # Hence i guess all should be returned
```
But for the Integral and Derivatives, i guess this is not correct i guess.
```
>>> Integral(x, (x, 0, a) ).atoms()
{x, a}                    # Hence all are free_symbols
>>> Integral(x, (x, 0, a)).free_symbols
{a}             # But {x, a} should be returned i guess.
````
As far as you said
> I also don't like how Symbol('+') is in the args

I think then the property  `expr == expr.func(*expr.args)` would not be satisfied
PS: I think there is some problem but not with `Limit` but rather with `Derivative` and `Integral` for `free_symbols` methods.
Please have a look, i am looking to try to solve the issue.

**Comment by gxyd** [comment]
this can be closed now. Fixed by #9581

**Comment by oscarbenjamin** [comment]
This is fixed now:
```
In [22]: x,a = symbols('x a') 
    ...: Limit(x,x,a).free_symbols                                                                                                             
Out[22]: {a}
```

It should have been closed by 39581 but wasn't because the key phrase "fixes #1234" wasn't spelled correctly.
